org.opentosca.planbuilder.type.plugin.dockercontainer/src/org/opentosca/planbuilder/type/plugin/dockercontainer/handler/BPMN4ToscaDockerContainerHandler.java

```diff
@@ -0,0 +1,165 @@
+import static org.opentosca.planbuilder.type.plugin.dockercontainer.handler.DockerContainerConstants.CONTAINER_IMAGE;
+import static org.opentosca.planbuilder.type.plugin.dockercontainer.handler.DockerContainerConstants.CONTAINER_IP;
+import static org.opentosca.planbuilder.type.plugin.dockercontainer.handler.DockerContainerConstants.CONTAINER_PORT;
+import static org.opentosca.planbuilder.type.plugin.dockercontainer.handler.DockerContainerConstants.CONTAINER_PORTS;
+import static org.opentosca.planbuilder.type.plugin.dockercontainer.handler.DockerContainerConstants.DOCKER_ENGINE_URL;
+import static org.opentosca.planbuilder.type.plugin.dockercontainer.handler.DockerContainerConstants.IMAGE_LOCATION;
+import static org.opentosca.planbuilder.type.plugin.dockercontainer.handler.DockerContainerConstants.PORT;
+import static org.opentosca.planbuilder.type.plugin.dockercontainer.handler.DockerContainerConstants.SSH_PORT;
+
+import java.util.LinkedList;
+import java.util.Map;
+
+import javax.xml.namespace.QName;
+
+import org.opentosca.planbuilder.bpmn4tosca.helpers.PropertyInitializer;
+import org.opentosca.planbuilder.model.plan.bpmn4tosca.BPMN4ToscaElement;
+import org.opentosca.planbuilder.model.plan.bpmn4tosca.BPMN4ToscaElementType;
+import org.opentosca.planbuilder.model.plan.bpmn4tosca.BPMN4ToscaTask;
+import org.opentosca.planbuilder.model.plan.bpmn4tosca.parameter.StringParameter;
+import org.opentosca.planbuilder.model.tosca.AbstractDeploymentArtifact;
+import org.opentosca.planbuilder.model.tosca.AbstractNodeTemplate;
+import org.opentosca.planbuilder.type.plugin.dockercontainer.utils.DockerContainerUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <p>
+ * <p>
+ * This class contains all the logic to add BPMN4Tosca Code which installs a
+ * docker container on a docker engine.
+ * </p>
+ * Copyright 2017 IAAS University of Stuttgart <br>
+ * <br>
+ *
+ * @author Alex Frank - st152404@stud.uni-stuttgart.de
+ *
+ */
+public final class BPMN4ToscaDockerContainerHandler {
+	private static final Logger LOGGER = LoggerFactory.getLogger(BPMN4ToscaDockerContainerHandler.class);
+
+	public boolean handle(LinkedList<BPMN4ToscaElement> parent, AbstractNodeTemplate nodeTemplate) {
+		LOGGER.debug("Handling Docker Container {}", nodeTemplate.getName());
+		boolean canHandle = false;
+		final String nodeId = nodeTemplate.getId();
+		final PropertyInitializer properties = new PropertyInitializer(nodeTemplate);
+		final Map<String, String> nodeProperties = properties.getPropertiesByTemplateId(nodeId);
+		final BPMN4ToscaTask bpmn4ToscaTask = new BPMN4ToscaTask();
+		final String containerPort = nodeProperties.get(CONTAINER_PORT);
+		final String port = nodeProperties.get(PORT);
+		final AbstractNodeTemplate dockerEngineTemplate = DockerContainerUtils.getDockerEngineNode(nodeTemplate);
+
+		if (containerPort != null && port != null) {
+			if (dockerEngineTemplate != null) {
```

Comment:
the level of indentation here could be reduced by inverting the condition and then early-returning like so:

```
if (containerPort == null || port == null) {
    LOGGER.error("Couldn't fetch Property variables ContainerPort or Port");
    return false;
}
if (dockerEngineTemplate == null) {
   LOGGER.error("Could not fetch DockerEngineNode from given Container");
    return false;
}
[...]
```

This is also known as "guard clause"